lte/gateway/python/magma/mobilityd/ip_address_man.py

```diff
@@ -333,21 +331,19 @@ def get_sid_ip_table(self) -> List[Tuple[str, ip_address]]:
 
     def get_ip_for_sid(self, sid: str) -> Optional[ip_address]:
         """ if ip is mapped to sid, return it, else return None """
-        with self._lock:
-            if sid in self._store.sid_ips_map:
-                if not self._store.sid_ips_map[sid]:
-                    raise AssertionError("Unexpected internal state")
-                else:
-                    return self._store.sid_ips_map[sid].ip
-            return None
+        if sid in self._store.sid_ips_map:
```

Comment:
Could we have a case where there is modification between 
`if sid in self._store.sid_ips_map:` and `if not self._store.sid_ips_map[sid]:`? where sid no longer exists? 
Or do we only read from redis the first time and not the second?

Reply:
Yeah I don't think we have a condition where we can hit this case, I updated the function to do the first check which will work the same.